raft/group0_tables/lang.hh

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+#pragma once
+
+#include <optional>
+#include <variant>
+
+#include "bytes.hh"
+
+
+namespace raft::group0_tables {
+
+// Represents 'SELECT value WHERE key = {key} FROM system.group0_kv_store;'.
+struct select_query {
+    bytes key;
+};
+
+// Represents 'UPDATE system.group0_kv_store SET value = {new_value} WHERE key = {key} [IF value = {value_condition}];'.
+// If value_condition is nullopt, the update is unconditional.
+struct update_query {
+    bytes key;
+    bytes new_value;
+    std::optional<bytes_opt> value_condition;
+};
```

Comment:
Note to reviewers: this was probably the simplest approach to implement a simple key-value store.

In the future, if we want to handle all of CQL and other frontends (Alternator etc.), we could do something like:
- for selects, store a `read_command`
- for conditional updates, store a `read_command` (to read the data needed to check the condition), and some kind of data structure that describes a program which walks over the fetched mutation and performs the checks. Something like cql3 restriction object but more low-level, operating on mutations; furthermore, a data structure which describes how to create a mutation
- for updates, store a `mutation`

But I think this simple kv-store is a good starter, because - as explained in the cover letter - it will let us work on implementing idempotency, snapshotting, and testing it with Jepsen or otherwise.

Reply:
But what's the plan? Throw it away later?

Reply:
The current language definition is a placeholder. We'll throw most of it away. Fortunately most of the work we want to do is independent of the language definition. Designing the language is only one of the tasks that need to be solved in this project. I didn't want to make it a blocker task that must be solved before any other tasks, hence we have this placeholder for the MVP, and we can now decide what to work on.

Reply:
As suggested by @kbr-, it will be done in the future

Reply:
Language definition is moved to the experimental subdirectory to emphasize that it is subject to change